cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMetricsHolder.java

```diff
@@ -0,0 +1,93 @@
+import io.strimzi.operator.common.MetricsProvider;
+import io.strimzi.operator.common.metrics.CertificateMetricKey;
+import io.strimzi.operator.common.metrics.MetricKey;
+import io.strimzi.operator.common.metrics.MetricsUtils;
+import io.strimzi.operator.common.metrics.OperatorMetricsHolder;
+import io.strimzi.operator.common.model.Labels;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Predicate;
+
+/**
+ * Operator metrics holder for Kafka assembly operator.
+ */
+public class KafkaAssemblyOperatorMetricsHolder extends OperatorMetricsHolder {
+    /**
+     * Metric name for certificate expiration timestamp in ms.
+     */
+    public static final String METRICS_CERTIFICATE_EXPIRATION_MS = METRICS_PREFIX + "certificate.expiration.timestamp.ms";
+
+    protected final Map<MetricKey, AtomicLong> certificateExpirationMap = new ConcurrentHashMap<>(1);
+
+    /**
+     * Constructs the operator metrics holder
+     *
+     * @param kind            Kind of the resources for which these metrics apply
+     * @param selectorLabels  Selector labels to select the controller resources
+     * @param metricsProvider Metrics provider
+     */
+    public KafkaAssemblyOperatorMetricsHolder(String kind, Labels selectorLabels, MetricsProvider metricsProvider) {
+        super(kind, selectorLabels, metricsProvider);
+    }
+
+
+    /**
+     * Time in milliseconds when the server certificate expiration timestamp in ms.
+     *
+     * @param clusterName   Name of the cluster
+     * @param namespace     Namespace of the resources being reconciled
+     * @return Metric gauge
+     */
+    public AtomicLong clusterCaCertificateExpiration(String clusterName, String namespace) {
+        return getGaugeLong(new CertificateMetricKey(kind, namespace, clusterName, CertificateMetricKey.Type.CLUSTER_CA),
+                METRICS_CERTIFICATE_EXPIRATION_MS, "Time in milliseconds when the certificate expires",
+                certificateExpirationMap, Tag.of("cluster", clusterName), Tag.of("type", CertificateMetricKey.Type.CLUSTER_CA.getDisplayName()));
+    }
+
+    /**
+     * Time in milliseconds when the client certificate expiration timestamp in ms.
+     *
+     * @param clusterName   Name of the cluster
+     * @param namespace     Namespace of the resources being reconciled
+     * @return Metric gauge
+     */
+    public AtomicLong clientCaCertificateExpiration(String clusterName, String namespace) {
+        return getGaugeLong(new CertificateMetricKey(kind, namespace, clusterName, CertificateMetricKey.Type.CLIENT_CA),
+                METRICS_CERTIFICATE_EXPIRATION_MS, "Time in milliseconds when the certificate expires",
+                certificateExpirationMap, Tag.of("cluster", clusterName), Tag.of("type", CertificateMetricKey.Type.CLIENT_CA.getDisplayName()));
+    }
```

Comment:
I was wondering if it makes sense to have just one method getting the certificate type from the outside. Wdyt?

Reply:
Sure, we could do that, was mainly due to maybe ones eventually wanted different number of tags or a different description for the different ones 😄